### DISCUSSION TITLE
running other console script from my code

### DISCUSSION THREAD

**Discussion by ws1088** [pex-tool/pex#1371] [Q&A] [answered]
Hi there, new to PEX!
I have written a python package which has a console script entry point. Somewhere in my code I have to call `pytest` which is a console_script from pytest package (I don't want to use pytest programmatic API to run pytest because of various reasons). In general, how do I run other console_scripts (I will be using `popen`)?
Thanks!

**Comment by jsirois** [comment, marked as answer]
You will only be able to do this with a `--venv` mode pex and the append or prepend option.

For example, using `--venv prepend` which prepends all console scripts found in the PEX to the PATH:
```console
$ cat src/exe.py 
import subprocess


def main():
  subprocess.check_call(["pytest", "--version"])
  subprocess.check_call(["which", "pytest"])

$ pex pytest --venv prepend -o my.pex -D src -e exe:main
$ ./my.pex 
pytest 6.2.4
/home/jsirois/.pex/venvs/efadb32f637132f9dc2de837e85c42205c95e130/2d1d404c3de23b1810386195be7410700b1feb14/bin/pytest
```

**Reply by jsirois** [reply]
From the help, this is:
```console
$ pex -h
...
PEX output options:
  Tailor the behavior of the emitted .pex file if -o is specified.

...
  --venv [{prepend,append}]
                        Convert the pex file to a venv before executing it. If 'prepend' or 'append' is specified, then all scripts and console scripts provided by distributions in the pex file will be added to the PATH in the corresponding position. If the the pex file will be run
                        multiple times under a stable runtime PEX_ROOT, the venv creation will only be done once and subsequent runs will enjoy lower startup latency. (default: False)
...
```

**Comment by ws1088** [comment]
thx. I think another solution I found out from the web is to do `system.check_call()` and provide an override to the script. But I guess that will run it twice...
